azure/02-managed-infra/index.ts

```diff
@@ -0,0 +1,29 @@
+import * as azure from "@pulumi/azure";
+import * as pulumi from "@pulumi/pulumi";
+import { config } from "./config";
+
+const name = pulumi.getProject();
+
+// Create a Virtual Network for the cluster
+const vnet = new azure.network.VirtualNetwork(name, {
+    resourceGroupName: config.resourceGroupName,
+    addressSpaces: ["10.2.0.0/16"],
+});
+
+// Create a Subnet for the cluster
+const subnet = new azure.network.Subnet(name, {
+    resourceGroupName: config.resourceGroupName,
+    virtualNetworkName: vnet.name,
+    addressPrefix: "10.2.1.0/24",
+});
```

Comment:
Could we add examples for using the default VPC and an existing VPC, much like we do in AWS?

https://github.com/pulumi/kubernetes-the-prod-way/blob/crosswalk/aws/02-managed-infra/index.ts

Reply:
Also see https://docs.microsoft.com/en-us/azure/aks/configure-azure-cni for more info and to ref in the docs portion about how Azure CNI manages pod networking for us